### DISCUSSION TITLE
No Module named 'health_azure'

### DISCUSSION THREAD

**Discussion by BK0x7C8** [microsoft/InnerEye-DeepLearning#734] [Q&A] [answered]
Hello folks,
I am currently setting up InnerEye on my work laptop (Win10) and so far the installation was successful. After running the HelloWorld segmentation model, I get the following error:

Traceback (most recent call last):
  File "InnerEye/ML/runner.py", line 30, in <module>
    from health_azure import AzureRunInfo, submit_to_azure_if_needed
ModuleNotFoundError: No module named 'health_azure'

Thanks for any kind of help!

Best,
Bjørn

**Comment by ant0nsc** [comment, marked as answer]
Hi @r00tdotexe , this sounds like the submodule for the `hi-ml` package is not available. Did you clone the repository with the `--recursive` flag as suggested in the Readme file? Can you check if there is a file `hi-ml/README.md` avaialable?

**Reply by BK0x7C8** [reply]
Hi @ant0nsc,
no, I copied the project as zip file and unpacked it to local drive.
The hi-ml folder is indeed empty. So I suggest, I will clone it as recommended and re-install.

**Reply by BK0x7C8** [reply]
@ant0nsc It works, Thanks for help!

**Comment by fepegar** [comment]
I am also having this issue. My submodules are correctly cloned.

```console
$ python -c "from InnerEye.ML.photometric_normalization import PhotometricNormalization"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/azureuser/git/InnerEye-DeepLearning/InnerEye/ML/photometric_normalization.py", line 13, in <module>
    from InnerEye.ML.config import PhotometricNormalizationMethod, SegmentationModelBase
  File "/home/azureuser/git/InnerEye-DeepLearning/InnerEye/ML/config.py", line 25, in <module>
    from InnerEye.ML.deep_learning_config import ModelCategory
  File "/home/azureuser/git/InnerEye-DeepLearning/InnerEye/ML/deep_learning_config.py", line 16, in <module>
    from InnerEye.Azure.azure_util import DEFAULT_CROSS_VALIDATION_SPLIT_INDEX, RUN_CONTEXT, is_offline_run_context
  File "/home/azureuser/git/InnerEye-DeepLearning/InnerEye/Azure/azure_util.py", line 16, in <module>
    from health_azure.utils import create_run_recovery_id, get_aml_run_from_run_id
ModuleNotFoundError: No module named 'health_azure'
```

Is this dependency specified anywhere? Cc @peterhessey

**Reply by peterhessey** [reply]
Interesting. No, `hi-ml-azure` is not specified as a dependency anywhere it seems as the code generally relies on being able to import from the git submodule, but `azure_util.py` does not for some unknown reason. I will create a separate issue for this and investigate more thoroughly ASAP. For others encountering this In the meantime, installing locally with pip (`pip install hi-ml-azure`) allows importing from `InnerEye.Azure.azure_util`.

**Reply by fepegar** [reply]
I suspect this hasn't been discovered before because people typically just call the runner, which I guess adds the libraries to the path.
